### ISSUE TITLE
stair-case edge effect when upsample from rgb_heatmap to original image size

### ISSUE THREAD

**Issue by xubo92** [google-coral/project-bodypix#39]
Hi @atvrager @dmitriykovalev @PeterMalkin @mbrooksx @hayatoy 

Thanks a lot for your contribution to this project!

I was trying to reproduce the results from this repo. When I tried to run bodypix.py using model "bodypix_mobilenet_v1_075_640_480_16_quant_decoder_edgetpu.tflite" to an image of persons, I got very obvious stair-case edge effect for the segmentation result. It looks as follows: left is original image while right is a combination of upsampled rgb_heatmap segment and original image.

![Image](https://github.com/user-attachments/assets/356de356-6f7a-409d-a19e-2002d5dacc30)

In this case, rgb_heatmap shape: (31, 41, 3), original image shape: (480, 640, 3), rescale factor: [15.483870967741936, 15.609756097560975, 1]

But the demo of this repo shows a very smoothed edge on segmented results. Would you kindly suggest on how the upsampling of rgb_heatmap is used to create the smooth edge effect like in the demo?

A similar issue was proposed here: https://github.com/google-coral/project-bodypix/issues/5

Thank you a lot!

**Comment by xubo92** [comment]
solved. the resize has to be done directly on the segmentation logits, rather on the uint8 pixel values.
